src/main/java/minegame159/meteorclient/systems/modules/misc/AutoWither.java

```diff
@@ -0,0 +1,149 @@
+
+    private Setting<Boolean> rotate = sgGeneral.add(new BoolSetting.Builder()
+            .name("rotate")
+            .description("Whether or not to rotate while building")
+            .defaultValue(true)
+            .build()
+    );
+
+    // Part 1 : Build wither directly infront of player (complete)
+    // Part 2 : Build wither based on where the player is looking
+    public AutoWither() {
+        super(Categories.World, "auto-wither", "Automatically builds withers.");
+    }
+
+    @EventHandler
+    private void onTick(TickEvent.Post event) {
+
+        // Check for soulsand and skull in hotbar
+        if (!hasEnoughMaterials()) {
+            error("(default)Not enough resources in hotbar");
+            toggle();
+            return;
+        }
+
+        // Find direction of player
+        // North, South, East, West
+        Direction dir = getDirection(mc.gameRenderer.getCamera().getYaw() % 360);
+
+
+        // Aligns player to center of block to avoid obstructing wither placement
+        PlayerUtils.centerPlayer();
+
+
+        // Check if we can place a wither
+        BlockPos blockPos = mc.player.getBlockPos();
+        blockPos = blockPos.offset(dir);
+
+        if (!isValidSpawn(blockPos, dir)) {
+            error("(default)Unable to spawn wither, obstructed by non air blocks");
+            toggle();
+            return;
+        }
+
+
+        // Build the wither
+        info("(default)Spawning wither");
+        spawnWither(blockPos, dir);
+        toggle();
+    }
+
+    private boolean hasEnoughMaterials() {
+        return (InvUtils.find(Items.SOUL_SAND).getCount() >= 4 || InvUtils.findInHotbar(Items.SOUL_SOIL).getCount() >= 4) &&
+            InvUtils.find(Items.WITHER_SKELETON_SKULL).getCount() >= 3;
+    }
+
+    private Direction getDirection(float yaw) {
+        if (yaw < 0) yaw += 360;
+
+        if (yaw >= 315 || yaw < 45) return Direction.SOUTH;
+        else if (yaw < 135) return Direction.WEST;
+        else if (yaw < 225) return Direction.NORTH;
+        else return Direction.EAST;
+    }
+
+    private boolean isValidSpawn(BlockPos blockPos, Direction direction) {
+
+        // Withers are 3x3x1
+
+        // Check if y > (255 - 3)
+        // Because withers are 3 blocks tall
+        if (blockPos.getY() > 252) return false;
+
+        // Determine width from direction
+        int widthX = 0;
+        int widthZ = 0;
+
+        if (direction == Direction.EAST || direction == Direction.WEST) widthZ = 1;
+        if (direction == Direction.NORTH || direction == Direction.SOUTH) widthX = 1;
+
+
+        // Check for non air blocks
+        for (int x = blockPos.getX() - widthX; x <= blockPos.getX() + widthX; x++) {
+            for (int z = blockPos.getZ() - widthZ; z <= blockPos.getZ(); z++) {
+                for (int y = blockPos.getY(); y <= blockPos.getY() + 2; y++) {
+                    if (mc.world.getBlockState(new BlockPos(x, y, z)).getBlock() != Blocks.AIR) return false;
```

Comment:
you should avoid making a new object this many times, just use a mutable object and set it here